File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/service/CollectTagService.java
package com.windea.demo.cloudcollect.core.service;

import com.windea.demo.cloudcollect.core.domain.entity.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * 收藏的标签的服务。
 */
public interface CollectTagService {
	/**
	 * 创建自己的标签。
	 */
	void create(CollectTag tag, User user);

	/**
	 * 删除自己的标签。
	 */
	void delete(Long id);

	/**
	 * 修改自己的标签（名字，概述）。
	 */
	void modify(Long id, CollectTag tag);

	/**
	 * 得到某一标签。
	 */
	CollectTag get(Long id);

	/**
	 * 分页得到某一标签的所有收藏。
	 */
	Page<Collect> getCollectPage(Long id, Pageable pageable);

	/**
	 * 得到某一标签的收藏数量。
	 */
	Long getCollectCount(Long id);

	/**
	 * 分页得到所有标签。
	 */
	Page<CollectTag> findAll(Pageable pageable);

	/**
	 * 分页查询某一用户的所有标签。
	 */
	Page<CollectTag> findByUser(Long userId, Pageable pageable);

	/**
	 * 根据名字分页模糊查询某一用户的所有标签。
	 */
	Page<CollectTag> findByUserAndName(Long userId, String name, Pageable pageable);

	/**
	 * 检查某一标签是否已存在。
	 */
	boolean exists(CollectTag tag);
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/service/CollectTagService.kt
package com.windea.demo.cloudcollect.core.service

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*

interface CollectTagService {
	/**创建自己的标签。*/
	fun create(tag: CollectTag, user: User)
	
	/**修改自己的标签。包括名字，概述*/
	fun modify(tag: CollectTag)
	
	/**删除自己的标签。*/
	fun deleteById(id: Long)
	
	/**根据id得到某一标签。*/
	fun findById(id: Long): CollectTag
	
	/**得到所有标签。*/
	fun findAll(pageable: Pageable): Page<CollectTag>
	
	/**根据名字和用户id模糊查询所有标签。*/
	fun findAllByNameContainsAndUserId(name: String, userId: Long, pageable: Pageable): Page<CollectTag>
	
	/**根据用户id查询所有标签。*/
	fun findAllByUserId(userId: Long, pageable: Pageable): Page<CollectTag>
	
	/**检查某一标签是否已存在。*/
	fun existsByNameAndUser(name: String, user: User): Boolean
	
	
	/**得到该标签的所有收藏。*/
	fun getCollectPage(id: Long, pageable: Pageable): Page<Collect>
}
